python/packages/core/agent_framework/_agents.py

```diff
@@ -999,6 +1002,140 @@ def get_new_thread(
             )
         return AgentThread(context_provider=self.context_provider)
 
+    def as_mcp_server(
+        self,
+        *,
+        server_name: str = "Agent",
+        version: str | None = None,
+        instructions: str | None = None,
+        lifespan: Callable[["Server[Any]"], AbstractAsyncContextManager[Any]] | None = None,
+        excluded_functions: str | Sequence[str] | None = None,
+        **kwargs: Any,
+    ) -> "Server[Any]":
+        """Create an MCP server from an agent instance.
+
+        This function automatically creates a MCP server from an agent instance, it uses the provided arguments to
+        configure the server and expose the agent's tools as individual MCP tools.
+
+        By default, all agent tools are exposed as MCP Tools, you can control this by using the
+        `excluded_functions` argument. These need to be set to the function name.
+
+        Args:
+            server_name: The name of the server.
+            version: The version of the server.
+            instructions: The instructions to use for the server.
+            lifespan: The lifespan of the server.
+            excluded_functions: The list of function names to exclude from the server.
+                if None, no functions will be excluded.
+            kwargs: Any extra arguments to pass to the server creation.
+
+        Returns:
+            The MCP server instance.
+        """
+        server_args: dict[str, Any] = {
+            "name": server_name,
+            "version": version,
+            "instructions": instructions,
+        }
+        if lifespan:
+            server_args["lifespan"] = lifespan
+        if kwargs:
+            server_args.update(kwargs)
+
+        if excluded_functions is not None and isinstance(excluded_functions, str):
+            excluded_functions = [excluded_functions]  # type: ignore
+
+        server: "Server[Any]" = Server(**server_args)  # type: ignore[call-arg]
+
+        # Get the agent's tools
+        tools_to_expose: list[AIFunction[Any, Any]] = []
```

Comment:
I was thinking this would expose the agent itself as a tool, not so much the underlying tools, if you would want that then why would you use AF, a simple FastMCP is much easier in that case, in SK this was also the case, where `as_mcp_server` on a agent would expose the agent itself not the tools, and there of course you could setup tools through kernel as a MCP server.

So for me this should be:

```suggestion
        tools_to_expose: list[AIFunction[Any, Any]] = [self.as_tool()]
```
and then inverse the `excluded_functions` to be `included_functions`

Reply:
I removed `excluded_functions` and didn't add `included_functions` because if `included_functions` is empty (default state), there will be no tools to invoke, so user has to manually specify all the functions to include, which is bad, because user already specified them when creating an agent. 

We can still add `excluded_functions` if we want a little bit more control, but that's also achievable with middleware.